waspc/src/Analyzer/TypeChecker/Monad.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE TupleSections #-}
+
+module Analyzer.TypeChecker.Monad where
```

Comment:
No `where ( ... )`? I am guessing you would want to hide at least `Bindings`?

Reply:
Solution for having properly interfaced modules but still exposing stuff for tests is to move the "private" stuff into `<ModuleName>.Internal`. So you could move those into `Analyzer.TypeChecker.Monad.Internal`. What I sometimes do, when I am very lazy, is created export list where I list both "public" and "private" stuff but then I group "private" stuff under the comment that says I exported it only for testing. But that is pretty bad, I believe the approach with `Internal` is what we should do.

Reply:
I get that, it is the same reason why I did what I described above, but I really don't like the idea of not having list of exports. So I would say imeither we do what I described above with special comments, or we do Internal module.

Reply:
I added a comment saying that `Bindings` and `runWithBound` were exported for testing.